lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -1987,40 +1987,37 @@ void TypeChecker::getPossibleTypesOfExpressionWithoutApplying(
     ExprTypeCheckListener *listener) {
   PrettyStackTraceExpr stackTrace(Context, "type-checking", expr);
 
+  ExprCleaner cleaner(expr);
```

Comment:
Oh so sad to see more of these but it's my fault since I haven't wrapped up the changes for the type map. :(